auth/Program.cs

```diff
@@ -15,18 +15,72 @@
  */
 // [START all]
 
+using Google.Apis.Auth.OAuth2;
 using Google.Apis.Storage.v1.Data;
 using Google.Cloud.Storage.V1;
 using System;
+using System.IO;
 
 namespace GoogleCloudSamples
 {
     public class AuthSample
     {
         const string usage = @"Usage:AuthSample <bucket_name>";
+        string _projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+
+        // [START auth_cloud_implicit]
+        void ImplicitAuth()
+        {
+            // If you don't specify credentials when constructing the client, the
+            // client library will look for credentials in the environment.
+            var storage = StorageClient.Create();
+            // Make an authenticated API request.
+            var buckets = storage.ListBuckets(_projectId);
+            foreach (var bucket in buckets)
+            {
+                Console.WriteLine(bucket.Name);
+            }
+        }
+        // [END auth_cloud_implicit]
+
+        void ExplicitAuth(string jsonPath)
+        {
+            // Explicitly use service account credentials by specifying the private key
+            // file.
+            GoogleCredential credential = null;
+            using (var jsonStream = new FileStream(jsonPath, FileMode.Open,
+                FileAccess.Read, FileShare.Read))
+            {
+                credential = GoogleCredential.FromStream(jsonStream);
+            }
+            var storage = StorageClient.Create(credential);
+            // Make an authenticated API request.
+            var buckets = storage.ListBuckets(_projectId);
+            foreach (var bucket in buckets)
+            {
+                Console.WriteLine(bucket.Name);
+            }
+        }
+
+        void ExplicitAuthComputeEngine()
+        {
+            // Explicitly use Compute Engine credentials. These credentials are
+            // available on Compute Engine, App Engine Flexible, and Container Engine.
+            var computeCredential = new ComputeCredential();
+            // How to instantiate?  This gives me a compile time error.

```

Comment:
I don't know that there's any way of doing that at the moment. If you're on ComputeEngine and nothing else is configured, `GoogleCredential.GetApplicationDefaultCredentials` will give you back a wrapped `ComputeCredential`, but I don't know of a way of doing it explicitly.

@chrisdunelm: Anything you know of?